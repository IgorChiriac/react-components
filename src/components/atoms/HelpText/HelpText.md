Additional information for input Components. This component is meant to be used with an input.

For a more complex component check `TextField`.

    <HelpText>Additional information</HelpText>
