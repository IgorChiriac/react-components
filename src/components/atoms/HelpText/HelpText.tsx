import React from 'react';
import styled from 'styled-components';
import * as colors from '../../../constants/colors';

export interface IHelpText extends React.HTMLAttributes<HTMLSpanElement> {}

const SHelpText = styled.span`
  color: ${colors.neutral.neutral700};
  line-height: 24px;
  display: block;
  font-size: 14px;
  font-weight: 400;
`;

const HelpText = (props: IHelpText) => <SHelpText {...props} />;

export default HelpText;
