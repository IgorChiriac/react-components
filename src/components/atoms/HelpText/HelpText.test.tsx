import React from 'react';
import { render } from 'react-testing-library';
import HelpText from './HelpText';

describe('<HelpText />', () => {
  it('renders the component with props', () => {
    const { container } = render(<HelpText>Additional info</HelpText>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
