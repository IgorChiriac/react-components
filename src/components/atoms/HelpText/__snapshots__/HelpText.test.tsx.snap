// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelpText /> renders the component with props 1`] = `
.c0 {
  color: #484E6C;
  line-height: 24px;
  display: block;
  font-size: 14px;
  font-weight: 400;
}

<span
  class="c0"
>
  Additional info
</span>
`;
